client/stack-editor/lib/index.coffee

```diff
@@ -140,9 +147,8 @@ module.exports = class StackEditorAppController extends AppController
 
     { computeController } = kd.singletons
 
-    EnvironmentFlux.actions.fetchAndUpdateStackTemplate templateId, (template) =>
-      @removeEditor template._id
-      @showView template
+    EnvironmentFlux.actions.fetchAndUpdateStackTemplate(templateId).then (template) =>
+      @editors[template._id]?.reloadRequired = yes
```

Comment:
Let's not modify the internal state of the editor here, instead you can write a method like `setReloadRequired` and it would handle binding necessary state variables. e.g a simple JS object (`@shouldReloadMap`), which has the id as key, and value as `true`(which is a dummy variable, simply if you have a key, just reload those editors, when needed). 

That way editors will still be **just** editors, and the state stays external. In the `RouteInfoHandled` event handler we can simply remove all the editors that should be removed (which we are reading it from `@shouldReloadMap`, or if you still want to make sure that we are only removing the selectedEditor instance, you can still keep the `return  unless @selectedEditor` line in the changes.